### PR TITLE
Add `all_of()` and `any_of()`

### PR DESCRIPTION
**Pull request by lionel-**
Depends on #124, r-lib/vctrs#615 and r-lib/vctrs#616.

### CODE REVIEW

File: R/select-helpers.R [hunk 1]
```diff
@@ -107,27 +120,34 @@ num_range <- function(prefix, range, width = NULL, vars = peek_vars()) {
...
 
-#' @export
 #' @rdname select_helpers
-#' @param ... One or more character vectors.
-one_of <- function(..., .vars = peek_vars()) {
-  keep <- compact(list(...))
+#' @param x A character vector.
+#' @inheritParams ellipsis::dots_empty
+#' @export
+all_of <- function(x, ..., vars = peek_vars()) {
```
Comment (hadley) [inline R/select-helpers.R]: Why do we allow dots here? Or conversely, why don't we use them?

Comment (lionel-) [reply R/select-helpers.R]: They are not allowed because of the ellipsis check. They are forcing `vars()` to be passed by name.

Comment (lionel-) [reply R/select-helpers.R]: It doesn't seem useful to take data by dots. This is a way of being prudent and conservative with the function signature.

Comment (hadley) [reply R/select-helpers.R]: :+1: can you make that clear in the news? (ie we’re being more conservative so the equivalent of `one_of(“a”, “b”)` is `all_of(c(“a”, “b”))`)

Comment (lionel-) [reply R/select-helpers.R]: Done in https://github.com/r-lib/tidyselect/commit/372eaf29f4244050d5e6a3b8308dddbccaa95247

**Pull request merged**